docs/manual/cluster-admin/how-to-customize-alerts.md

```diff
@@ -0,0 +1,155 @@
+
+```yaml
+alert-manager:
+  port: 9093 # optional, do not change this if you do not want to change the port alert-manager is listening on
+  email_configs: # email-notification will only be enabled when this field is not empty
+    receiver: your_addr@example.com
+    smtp_url: smtp.office365.com:587
+    smtp_from: alert_sender@example.com
+    smtp_auth_username: alert_sender@example.com
+    smtp_auth_password: password_for_alert_sender
+  webhook_configs: # webhook-actions will only be enabled when  this field is not empty
+    port: 9095 # optional, do not change this if you do not want to change the port alert-handler is listening on
+    pai_bearer_token: 'your_application_token_for_rest_server'
+```
+
+### How to Match Alerts and Actions
+
+The matching rules of alerts and actions are defined in [alert-configmap.yaml.template](https://github.com/microsoft/pai/blob/master/src/alert-manager/deploy/alert-configmap.yaml.template):
+
+``` yaml
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: alertmanager
+data:
+  config.yml: |-
+{% set email_configs = cluster_cfg["alert-manager"]["email_configs"] %}
+{% set webhook_configs = cluster_cfg["alert-manager"]["webhook_configs"] %}
+    global:
+      resolve_timeout: 5m
+{% if cluster_cfg["alert-manager"]["email-configured"] %}
+      smtp_smarthost: {{ email_configs["smtp_url"] }}
+      smtp_from: {{ email_configs["smtp_from"] }}
+      smtp_auth_username: {{ email_configs["smtp_auth_username"] }}
+      smtp_auth_password: {{ email_configs["smtp_auth_password"] }}
+{% endif %}
+    
+    templates:
+    - "/etc/alertmanager/template/*.tmpl"
+
+    route:
+      receiver: pai-email
+      group_wait: 30s
+      group_interval: 5m
+      repeat_interval: {{ cluster_cfg["alert-manager"]["repeat-interval"] }}
+      group_by: [alertname, alertstate]
+      routes:
+      - receiver: pai-email-and-stop-job
+        match: 
+          alertname: PAIJobGpuPercentLowerThan0_3For1h
+    
+    receivers:
+    - name: "pai-email"
+{% if cluster_cfg["alert-manager"]["email-configured"] %}
+      email_configs:
+        - to: {{ email_configs["receiver"] }}
+          send_resolved: true
+          html: '{{ "{{" }} template "email.pai.html" . {{ "}}" }}'
+          headers:
+            subject: '{{ cluster_cfg["cluster"]["common"]["cluster-id"] }}: {{ "{{" }} template "__subject" . {{ "}}" }}'
+{% endif %}
+    - name: "pai-email-and-stop-job"
+{% if cluster_cfg["alert-manager"]["email-configured"] %}
+      email_configs:
+        - to: {{ email_configs["receiver"] }}
+          send_resolved: true
+          html: '{{ "{{" }} template "email.pai.html" . {{ "}}" }}'
+          headers:
+            subject: '{{ cluster_cfg["cluster"]["common"]["cluster-id"] }}: {{ "{{" }} template "__subject" . {{ "}}" }}'
+{% endif %}
+{% if cluster_cfg["alert-manager"]["webhook-configured"] %}
+     # webhook_configs:
+     #   - url: 'http://localhost:{{ webhook_configs["port"] }}/alert-handler/stop-job'
```

Comment:
Maybe we can send stop email to user at webhook. Seems more flexible. We can get user profile through PAI API.  We can do this when you have time.